API: Failing test when it runs alone

<!--
Please use this template to create your bug report. By providing as much info as possible you help us understand the issue, reproduce it and resolve it for you quicker. Therefore take a couple of extra minutes to make sure you have provided all info needed.

PROTIP: record your screen and attach it as a gif to showcase the issue.

- Questions should be posted to: https://community.grafana.com
- Use query inspector to troubleshoot issues: https://bit.ly/2XNF6YS
- How to record and attach gif: https://bit.ly/2Mi8T6K
-->

**What happened**:
The following test panics when it runs 
<details>
<summary>Failing test</summary>

```
go test -v ./pkg/api -run TestFoldersAPIEndpoint
=== RUN   TestFoldersAPIEndpoint
=== RUN   TestFoldersAPIEndpoint/Given_a_correct_request_for_creating_a_folder
=== RUN   TestFoldersAPIEndpoint/Given_a_correct_request_for_creating_a_folder/When_calling_POST_on_/api/folders
--- FAIL: TestFoldersAPIEndpoint (0.14s)
    --- FAIL: TestFoldersAPIEndpoint/Given_a_correct_request_for_creating_a_folder (0.14s)
        --- FAIL: TestFoldersAPIEndpoint/Given_a_correct_request_for_creating_a_folder/When_calling_POST_on_/api/folders (0.14s)
panic: no guardian factory implementation provided [recovered]
        panic: no guardian factory implementation provided

goroutine 150 [running]:
testing.tRunner.func1.2({0x5b6c2a0, 0x64f7940})
        /usr/local/opt/go/libexec/src/testing/testing.go:1389 +0x24e
testing.tRunner.func1()
        /usr/local/opt/go/libexec/src/testing/testing.go:1392 +0x39f
panic({0x5b6c2a0, 0x64f7940})
        /usr/local/opt/go/libexec/src/runtime/panic.go:838 +0x207
github.com/grafana/grafana/pkg/services/guardian.glob..func1({0xc0011062d0?, 0x6512ca8?}, 0xc0005ddb90?, 0xc0005fe0e0?, 0x38?)
        /Users/josefk/go/src/github.com/grafana/grafana/pkg/services/guardian/guardian.go:54 +0x27
github.com/grafana/grafana/pkg/api.(*HTTPServer).CreateFolder(0xc00008d200, 0xc000958000)
        /Users/josefk/go/src/github.com/grafana/grafana/pkg/api/folder.go:83 +0x392
github.com/grafana/grafana/pkg/api.createFolderScenario.func1.1(0xc000958000)
        /Users/josefk/go/src/github.com/grafana/grafana/pkg/api/folder_test.go:153 +0x1e9
github.com/grafana/grafana/pkg/api/routing.Wrap.func1(0xc0005ddb90?)
        /Users/josefk/go/src/github.com/grafana/grafana/pkg/api/routing/routing.go:17 +0x22
github.com/grafana/grafana/pkg/api/response.wrap_handler.func3({0x5cae600?, 0x5e17d40?}, 0x6506700?)
        /Users/josefk/go/src/github.com/grafana/grafana/pkg/api/response/web_hack.go:39 +0x73
net/http.HandlerFunc.ServeHTTP(0xc000ce59f0?, {0x650f778?, 0xc000d8f630?}, 0xc000ce5a48?)
        /usr/local/opt/go/libexec/src/net/http/server.go:2084 +0x2f
github.com/grafana/grafana/pkg/web.(*Context).run(0xc001085b60)
        /Users/josefk/go/src/github.com/grafana/grafana/pkg/web/context.go:60 +0xaa
github.com/grafana/grafana/pkg/web.(*Context).Next(...)
        /Users/josefk/go/src/github.com/grafana/grafana/pkg/web/context.go:55
github.com/grafana/grafana/pkg/web.(*Macaron).UseMiddleware.func1({0x650f778?, 0xc000d8f630?}, 0xc00121a600)
        /Users/josefk/go/src/github.com/grafana/grafana/pkg/web/macaron.go:139 +0x1a5
net/http.HandlerFunc.ServeHTTP(0x65067a0?, {0x650f778?, 0xc000d8f630?}, 0x2?)
        /usr/local/opt/go/libexec/src/net/http/server.go:2084 +0x2f
github.com/grafana/grafana/pkg/web.Renderer.func1.1({0x650f778, 0xc000d8f630}, 0xc00121a600)
        /Users/josefk/go/src/github.com/grafana/grafana/pkg/web/render.go:44 +0x1e7
net/http.HandlerFunc.ServeHTTP(0x4016526?, {0x650f778?, 0xc000d8f630?}, 0x40516d2?)
        /usr/local/opt/go/libexec/src/net/http/server.go:2084 +0x2f
github.com/grafana/grafana/pkg/web.(*Context).run(0xc001085b60)
        /Users/josefk/go/src/github.com/grafana/grafana/pkg/web/context.go:60 +0xaa
github.com/grafana/grafana/pkg/web.(*Router).Handle.func1({0x6511788, 0xc000f34340}, 0x5f450d3?, 0x5f58293?)
        /Users/josefk/go/src/github.com/grafana/grafana/pkg/web/router.go:156 +0x22f
github.com/grafana/grafana/pkg/web.(*Router).ServeHTTP(0xc00111e840, {0x6511788, 0xc000f34340}, 0xc00121a400)
        /Users/josefk/go/src/github.com/grafana/grafana/pkg/web/router.go:213 +0x1b6
github.com/grafana/grafana/pkg/web.(*Macaron).ServeHTTP(0xc00097ea50?, {0x6511788?, 0xc000f34340?}, 0xc00121a400?)
        /Users/josefk/go/src/github.com/grafana/grafana/pkg/web/macaron.go:170 +0xf1
github.com/grafana/grafana/pkg/api.(*scenarioContext).exec(...)
        /Users/josefk/go/src/github.com/grafana/grafana/pkg/api/common_test.go:175
github.com/grafana/grafana/pkg/api.callCreateFolder(0xc001228090)
        /Users/josefk/go/src/github.com/grafana/grafana/pkg/api/folder_test.go:134 +0xb6
github.com/grafana/grafana/pkg/api.TestFoldersAPIEndpoint.func1.1(0xc001228090)
        /Users/josefk/go/src/github.com/grafana/grafana/pkg/api/folder_test.go:36 +0x2b
github.com/grafana/grafana/pkg/api.createFolderScenario.func1(0xc0011284e0?)
        /Users/josefk/go/src/github.com/grafana/grafana/pkg/api/folder_test.go:158 +0x2cc
testing.tRunner(0xc001128820, 0xc0008241c0)
        /usr/local/opt/go/libexec/src/testing/testing.go:1439 +0x102
created by testing.(*T).Run
        /usr/local/opt/go/libexec/src/testing/testing.go:1486 +0x35f
FAIL    github.com/grafana/grafana/pkg/api      0.794s
FAIL
```
</details>

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:
The test misses mocking the guarding service

**Environment**:
- Grafana version:
- Data source type & version:
- OS Grafana is installed on:
- User OS & Browser:
- Grafana plugins:
- Others:

--- a/pkg/api/folder_test.go
+++ b/pkg/api/folder_test.go
@@ -12,6 +12,8 @@ import (
 	"github.com/grafana/grafana/pkg/models"
 	"github.com/grafana/grafana/pkg/services/dashboards"
 	"github.com/grafana/grafana/pkg/services/featuremgmt"
+	"github.com/grafana/grafana/pkg/services/guardian"
+	"github.com/grafana/grafana/pkg/services/sqlstore/mockstore"
 	"github.com/grafana/grafana/pkg/setting"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/mock"
@@ -137,6 +139,14 @@ func callCreateFolder(sc *scenarioContext) {
 func createFolderScenario(t *testing.T, desc string, url string, routePattern string, folderService dashboards.FolderService,
 	cmd models.CreateFolderCommand, fn scenarioFunc) {
 	t.Run(fmt.Sprintf("%s %s", desc, url), func(t *testing.T) {
+		aclMockResp := []*models.DashboardAclInfoDTO{}
+		dashSvc := &dashboards.FakeDashboardService{}
+		dashSvc.On("GetDashboardAclInfoList", mock.Anything, mock.AnythingOfType("*models.GetDashboardAclInfoListQuery")).Run(func(args mock.Arguments) {
+			q := args.Get(1).(*models.GetDashboardAclInfoListQuery)
+			q.Result = aclMockResp
+		}).Return(nil)
+		store := mockstore.NewSQLStoreMock()
+		guardian.InitLegacyGuardian(store, dashSvc)
 		hs := HTTPServer{
 			Cfg:           setting.NewCfg(),
 			folderService: folderService,

